src/tests/mod.rs

```diff
@@ -63,23 +62,37 @@ fn copy_folder(src: &Path, dst: &Path) {
 }
 
 /// Checks if the file updates returned by piranha are as expected.
-fn check_result(output_summaries: Vec<PiranhaOutputSummary>, path_to_expected: PathBuf) {
+fn check_result(path_to_codebase: &Path, path_to_expected: &Path) {
   let mut all_files_match = true;
 
-  for summary in &output_summaries {
-    let updated_file_name = &summary
-      .path()
-      .file_name()
-      .and_then(|f| f.to_str().map(|x| x.to_string()))
-      .unwrap();
-    let expected_file_path = find_file(&path_to_expected, updated_file_name);
-    let expected_content = read_file(&expected_file_path).unwrap();
+  let count_files = |path: &Path| {
+    let mut count = 0;
+    for dir_entry in fs::read_dir(path).unwrap() {
+      if dir_entry.unwrap().path().is_file() {
+        count += 1;
+      }
+    }
+    count
+  };
+
+  assert_eq!(count_files(path_to_codebase), count_files(path_to_expected));
```

Comment:
Won't this break for cases where we _are_ deleting empty files? Don't we want to be able to test for that too?

Reply:
https://github.com/uber/piranha/blob/73e84d1c4829196032584d281621a38c41d6057b/src/models/unit_tests/source_code_unit_test.rs#L208 

We test this functionality here in this "unit test" (Atleast thats what I called this before). 
This unit test does kinda what `create_rewrite_rules` macro does. I plan to move these unit test to the macro.
These unit tests run via this `execute_persist_in_temp` method which is equivalent to `create_rewrite_rules` now.

Reply:
Ok, still think it might make sense to test this at least once in our "integration" tests to verify that it indeed counts things correctly when `|input/| < |expected/|`, but at this point I am fairly convinced this isn't _required_, sorry for the confusion 🙂